pkg/e2e/operators/cluster-autoscaler-operator.go

```diff
@@ -62,12 +62,25 @@ var _ = g.Describe("[Feature:Operators] Cluster autoscaler cluster operator shou
 				glog.Errorf("error querying api for OperatorStatus object: %v, retrying...", err)
 				return false, nil
 			}
+			ready := false
 			if available := cvoresourcemerge.FindOperatorStatusCondition(clusterOperator.Status.Conditions, osconfigv1.OperatorAvailable); available != nil {
 				if available.Status == osconfigv1.ConditionTrue {
-					return true, nil
+					ready = true
+				} else {
+					ready = false
 				}
 			}
-			return false, nil
+			if progressing := cvoresourcemerge.FindOperatorStatusCondition(clusterOperator.Status.Conditions, osconfigv1.OperatorProgressing); progressing != nil {
+				if progressing.Status == osconfigv1.ConditionTrue {
+					ready = false
+				}
+			}
+			if failing := cvoresourcemerge.FindOperatorStatusCondition(clusterOperator.Status.Conditions, osconfigv1.OperatorFailing); failing != nil {
+				if failing.Status == osconfigv1.ConditionTrue {
+					ready = false
+				}
+			}
+			return ready, nil
```

Comment:
and here return `return true, nil` and remove `ready` var completely.

Reply:
@ingvagabund done